### ISSUE TITLE
Post json string

### ISSUE THREAD

**Issue by bologer** [linslin/Yii2-Curl#68]
Sometimes it is useful to post simple json string: 

```
curl http://youdomain.com/api --data '["whatever here", "again"]'
```
In this case this would not work: 
```php
/**
 * Set get params
 *
 * @param array $params
 * @return $this
 */
public function setPostParams($params)
{
    if (is_array($params)) {
        $this->setOption(
            CURLOPT_POSTFIELDS,
            http_build_query($params)
        );
    }
    return $this;
}
```

it is required to change it to this to make method accept plain strings as well:
```php
/**
 * Set get params as array or as plain string, such as JSON.
 *
 * @param array|string $params
 * @return $this
 */
public function setPostParams($params)
{
    $this->setOption(
        CURLOPT_POSTFIELDS,
        is_array($params) ? http_build_query($params) : $params
    );
    return $this;
}
```

**Comment by linslin** [comment]
Thanks m8,

so you want to do somethink like:

`curl -H "Content-Type: application/json" -X POST -d '{"data":"myData","someMoreDate":"xyz"}' http://localhost:3000/api/endpoint`

Yea this sounds like a common case to me. Let me check this in the next days.

**Comment by bologer** [comment]
@linslin , this is true that you may do plain cURL, but what is the point of the library then? It is better to implement here and use at a feature.

**Comment by linslin** [comment]
My post is just an example of what you want to do with yii2/curl. So, yes. I will implement this feature in the next release. Good point!

**Comment by bologer** [comment]
@linslin thanks

**Comment by linslin** [comment]
Its released in 1.2.1 

```
// POST RAW JSON
$curl = new curl\Curl();
$response = $curl->setRawPostData(
     json_encode[
        'key' => 'value',
        'secondKey' => 'secondValue'
     ])
     ->post('http://example.com/');
```

```
// POST RAW XML
$curl = new curl\Curl();
$response = $curl->setRawPostData('<?xml version="1.0" encoding="UTF-8"?><someNode>Test</someNode>')
     ->post('http://example.com/');
```
